src/extensions/layout-selector/index.js

```diff
@@ -92,6 +93,28 @@ class LayoutSelector extends Component {
 		this.useTemplateLayout = this.useTemplateLayout.bind( this );
 		this.useEmptyTemplateLayout = this.useEmptyTemplateLayout.bind( this );
 		this.renderContent = this.renderContent.bind( this );
+		this.detectImageBlocks = this.detectImageBlocks.bind( this );
+		this.uploadExternalImages = this.uploadExternalImages.bind( this );
+	}
+
+	componentDidUpdate( prevProps ) {
+		if ( prevProps.clientIds.length === 0 && this.props.clientIds.length !== 0 ) {
+			this.detectImageBlocks( this.props.clientIds )
+				.filter(
+					( block ) => {
+						if ( typeof block === 'undefined' || isEmpty( Object.values( block )[ 0 ] ) ) {
```

Comment:
@jrtashjian I think this could easily by a one liner and still be readable. Just a simple return statement with the condition.

Reply:
Something like this @jonathanbardo?
```javascript
.filter( ( block ) => ( typeof block === 'undefined' || isEmpty( Object.values( block )[ 0 ] ) ) ? false : true )
```

Reply:
Yes exactly

Reply:
Fixed in 7c959cb